core/src/collections.rs

```diff
@@ -138,53 +138,74 @@ impl MongoStatement for MongoCollections {
     // Move the cursor to the next CollectionSpecification.
     // When cursor is exhausted move to next database in list
     // Return true if moving was successful, false otherwise.
-    fn next(&mut self, _: Option<&MongoConnection>) -> Result<(bool, Option<Error>)> {
+    fn next(&mut self, _: Option<&MongoConnection>) -> Result<(bool, Option<Vec<Error>>)> {
         if self.current_database_index.is_none() {
             if self.collections_for_db_list.is_empty() {
                 return Ok((false, None));
             }
             self.current_database_index = Some(0);
         }
         loop {
-            while self
-                .collections_for_db_list
-                .get_mut(self.current_database_index.unwrap())
-                .unwrap()
-                .collection_list
-                .advance()
-                .map_err(Error::Mongo)?
-            {
-                let collection = self
+            let mut e: Vec<Error> = vec![];
+            loop {
+                match self
                     .collections_for_db_list
-                    .get(self.current_database_index.unwrap())
+                    .get_mut(self.current_database_index.unwrap())
                     .unwrap()
                     .collection_list
-                    .deserialize_current()
-                    .map_err(Error::Mongo)?;
-                // Filter the collection matching the types and names specified by the user
-                if (self.table_types_filter.is_none()
-                    || self
-                        .table_types_filter
-                        .as_ref()
-                        .unwrap()
-                        .contains(&collection.collection_type))
-                    && (self.collection_name_filter.is_none()
-                        || self
-                            .collection_name_filter
-                            .as_ref()
-                            .unwrap()
-                            .is_match(&collection.name))
+                    .advance()
```

Comment:
Could you use https://doc.rust-lang.org/std/result/enum.Result.html#method.or_else to add errors to the vector instead having to use match blocks?